### DISCUSSION TITLE
How to correctly delete fields that contain non-literary characters?

### DISCUSSION THREAD

**Discussion by bmday** [vectordotdev/vrl#992] [Q&A] [answered]
Hi!
We use Vector with Remap transform for Kubernetes logs and we need to delete some fields from processing data, e.g.:

```
    transforms:
      kubernetes_logs_transformer:
        type: remap
        inputs:
          - kubernetes_logs
        source: |-
            del(.kubernetes.labels.pod-template-hash) - no work
            del(.kubernetes.node_labels.beta.kubernetes.io/arch)  - no work
            del(.kubernetes.node_labels.kubernetes.io/arch)  - no work
            del(.kubernetes.container_name) - It work
```
It was expected that these operations would remove fields from the output, but it did not work. This gives a syntax error, which is logical if you pay attention to the documentation for Path: https://vector.dev/docs/reference/vrl/expressions/#path

> **path_segments** denote a segment of a nested path. Each segment must be delimited by a . character and only contain alpha-numeric characters and _ (a-zA-Z0-9_). Segments that contain characters outside of this range must be quoted.

![image](https://github.com/user-attachments/assets/90c80258-e55f-48b3-8e88-1d893aae0a7c)

Аfter adding the quotes, the sitaxis errors disappeared, but had no effect. The fields are not deleted as expected.

```
    transforms:
      kubernetes_logs_transformer:
        type: remap
        inputs:
          - kubernetes_logs
        source: |-
            del(.kubernetes.labels."pod-template-hash") - It work
            del(.kubernetes.node_labels.beta.kubernetes."io/arch") - No effect
            del(.kubernetes.node_labels.kubernetes."io/arch") - No effect
            del(.kubernetes.container_name) - It work
```
How to correctly delete fields that contain non-literary characters?

**Comment by jszwedko** [comment]
Try `del(.kubernetes.node_labels."kubernetes.io/arch")`. The need to quote path segments with special characters is mentioned here: https://vector.dev/docs/reference/vrl/expressions/#path

**Reply by bmday** [reply]
Thanks!
I try
```
del(.kubernetes.node_labels.beta.kubernetes."io/arch")
del(."kubernetes.node_labels.beta.kubernetes.io/os")
```
but it didn't work either. The fields continue to be sent to the output.

**Reply by bmday** [reply]
UPD.

Experimentally, it was possible to establish that the following configuration works:
```
            del(.kubernetes.namespace_labels."kubernetes.io/metadata.name")
            del(.kubernetes.node_labels."beta.kubernetes.io/arch")
            del(.kubernetes.node_labels."beta.kubernetes.io/os")
```
This is certainly a good result, but it was not easy to understand in advance that this is how it works, and not otherwise.

Thank you, @jszwedko

**Reply by jszwedko** [reply, marked as answer]
👍 all keys under `node_labels` are "flat" and so should be quoted.
